README.md
# Mattermost Cookbook

[![Build Status](https://travis-ci.org/ist-dsi/mattermost-cookbook.svg?branch=master)](https://travis-ci.org/ist-dsi/mattermost-cookbook) [![Cookbook Version](https://img.shields.io/cookbook/v/mattermost-cookbook.svg)](https://supermarket.chef.io/cookbooks/mattermost-cookbook)

## Overview

This cookbook installs and configures [Mattermost, the opensource Slack alternative](http://www.mattermost.org/).

### Supported Mattermost Versions

| Mattermost Version | Cookbook Version |
|:------------------:|:----------------:|
| 5.26.2             | v5.26.2          |
| 5.25.7             | v5.25.7          |
| 5.24.3             | v5.24.3          |
| 5.23.2             | v5.23.2          |
| 5.22.3             | v5.22.3          |
| 5.21.0             | v5.21.0          |
| 5.20.2             | v5.20.2          |
| 5.19.3             | v5.19.3          |
| 5.18.2             | v5.18.2          |
| 5.17.1             | v5.17.1          |
| 5.16.3             | v5.16.3          |
| 5.15.1             | v5.15.1          |
| 5.14.3             | v5.14.3          |
| 5.13.3             | v5.13.3          |
| 5.12.2             | v5.12.2          |
| 5.12.0             | v5.12.0          |
| 5.11.1             | v5.11.1          |
| 5.10.0             | v5.10.0          |
| 5.9.0              | v5.9.0           |
| 5.8.0              | v5.8.0           |
| 5.7.2              | v5.7.2           |
| 5.7.0              | v5.7.0           |
| 5.6.3              | v5.6.3           |
| 5.5.1              | v5.5.1           |
| 5.5.0              | v5.5.0           |
| 5.4.0              | v5.0.1           |
| 5.4.0              | v5.0.0           |
| 5.3.1              | v4.0.1           |
| 5.2.1              | v3.0.0           |
| 5.1.1              | v2.1.1           |
| 5.0.1              | v2.0.0           |
| 4.10.1             | v1.6.10          |

For older versions see this cookbook history at [Supermarket](https://supermarket.chef.io/cookbooks/mattermost-cookbook).

## Requirements

### Platforms

The following platforms and versions are tested and supported using [test-kitchen](http://kitchen.ci/)  

* Ubuntu 18 LTS
* Debian 9
* Debian 10
* CentOS 7

### Chef

* Chef 12.19+

### Dependencies

* [tar](https://supermarket.chef.io/cookbooks/tar)

## Usage

Using this cookbook is relatively straightforward. It is recommended to create a project or organization specific [wrapper cookbook](https://www.chef.io/blog/2013/12/03/doing-wrapper-cookbooks-right/) and add the desired recipes (for example, to setup a database) to the run list of a node.

### Attributes

All settings are the [default settings](https://docs.mattermost.com/administration/config-settings.html). Make sure to review them all and change what's necessary.

## License and Authors

* Author:: Simão Martins    <simao.martins@tecnico.ulisboa.pt>
* Author:: Simão Silva      <simao.silva@tecnico.ulisboa.pt>
* Author:: David Duarte     <david.duarte@tecnico.ulisboa.pt>
* Author:: Kyle Corupe      <kyle.corupe@gmail.com>

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.


CHANGELOG.md
# Mattermost Cookbook Changelog

This file is used to list changes made in each version of the Mattermost cookbook.
## v5.26.2 (2021.04.20)

- Update to Mattermost 5.26.2
## v5.25.7 (2021.04.20)

- Update to Mattermost 5.25.7
## v5.24.3 (2021.04.20)

- Update to Mattermost 5.24.3
## v5.23.2 (2021.04.20)

- Update to Mattermost 5.23.2
## v5.22.3 (2021.04.20)

- Update to Mattermost 5.22.3
## v5.21.0 (2021.04.20)

- Update to Mattermost 5.21.0
## v5.20.2 (2021.04.20)

- Update to Mattermost 5.20.2
## v5.19.3 (2021.04.20)

- Update to Mattermost 5.19.3
## v5.18.2 (2021.04.20)

- Update to Mattermost 5.18.2

## v5.17.1 (2019.12.16)

- [#57](https://github.com/ist-dsi/mattermost-cookbook/issues/57) Update to Mattermost 5.17.1

## v5.16.3 (2019.12.13)

- Now supports Mattermost 5.16.3

## v5.15.1 (2019.10.15)

- [#55](https://github.com/ist-dsi/mattermost-cookbook/issues/55) Update to Mattermost 5.15.1

## v5.14.3 (2019.10.02) 

- Now supports Mattermost 5.14.3

## v5.13.3 (2019.10.01) 

- Now supports Mattermost 5.13.3

## v5.12.2 (2019.07.04)

- [#48](https://github.com/ist-dsi/mattermost-cookbook/issues/48) Update to Mattermost 5.12.2

## v5.12.0 (2019.06.24)

- [#47](https://github.com/ist-dsi/mattermost-cookbook/issues/47) Update to Mattermost 5.12.0

## v5.11.1 (2019.06.24)

- Now supports Mattermost 5.11.1

## v5.10.0 (2019.05.13)

- [#45](https://github.com/ist-dsi/mattermost-cookbook/issues/45) Update to Mattermost 5.10.0

## v5.9.0 (2019.03.25)

- [#44](https://github.com/ist-dsi/mattermost-cookbook/issues/44) Update to Mattermost 5.9.0

## v5.8.0 (2019.03.11)

- Now supports Mattermost v5.8.0

## v5.7.2 (2019.03.11)

- [#43](https://github.com/ist-dsi/mattermost-cookbook/issues/43) Update to Mattermost 5.7.2

## v5.7.0 (2019.01.17)

- Now supports Mattermost v5.7.0

## v5.6.3 (2019.01.17)

- [#42](https://github.com/ist-dsi/mattermost-cookbook/issues/42) Update to Mattermost 5.6.3

## v5.5.1 (2018.12.06)

- [#41](https://github.com/ist-dsi/mattermost-cookbook/issues/41) Update to Mattermost 5.5.1

## v5.5.0 (2018.11.27)

- [#40](https://github.com/ist-dsi/mattermost-cookbook/issues/40) Update to Mattermost 5.5.0

## v5.0.1 (2018.10.26)

- Fix setting CAP_NET_BIND_SERVICE
- Remove extra coma from `config.json`

## v5.0.0 (2018.10.17)

- [#39](https://github.com/ist-dsi/mattermost-cookbook/issues/39) Update to Mattermost 5.4.0

## v4.0.1 (2018.10.12)

- Create plugins directory.

## v4.0.0 (2018.09.24)

- [#38](https://github.com/ist-dsi/mattermost-cookbook/issues/38) Update to Mattermost 5.3.1

## v3.0.0 (2018.08.27)

- [#37](https://github.com/ist-dsi/mattermost-cookbook/issues/37) Update to Mattermost 5.2.1

## v2.1.1 (2018.08.13)

- [#34](https://github.com/ist-dsi/mattermost-cookbook/issues/34) Update to Mattermost 5.1.1
- [#35](https://github.com/ist-dsi/mattermost-cookbook/pull/35) Add `download_dir` option to `tar_extract` resource.

## v2.1.0 (2018.07.16)

- Update settings for Mattermost 5.1.0

## v2.0.0 (2018.07.13)

- [#32](https://github.com/ist-dsi/mattermost-cookbook/issues/32) Update settings for Mattermost 5.0.1

## v1.6.10 (2018.06.14)

- [#30](https://github.com/ist-dsi/mattermost-cookbook/issues/30) Update settings for Mattermost 4.10.1
- Replace [ark](https://supermarket.chef.io/cookbooks/ark) cookbook with [tar](https://supermarket.chef.io/cookbooks/tar)

## v1.6.9 (2018.05.14)

- [#26](https://github.com/ist-dsi/mattermost-cookbook/issues/24) Update settings for Mattermost 4.9.2

## v1.6.8 (2018.04.17)

- [#24](https://github.com/ist-dsi/mattermost-cookbook/issues/24) Update settings for Mattermost 4.9.0

## v1.6.7 (2018.04.16)

- [#23](https://github.com/ist-dsi/mattermost-cookbook/issues/23) Update settings for Mattermost 4.8.1

## v1.6.6 (2018.03.23)

- [#22](https://github.com/ist-dsi/mattermost-cookbook/issues/22) Update settings for Mattermost 4.8.0

## v1.6.5 (2018.03.13)

- [#21](https://github.com/ist-dsi/mattermost-cookbook/issues/21) Update settings for Mattermost 4.7.3

## v1.6.4 (2018.02.21)

- [#20](https://github.com/ist-dsi/mattermost-cookbook/issues/20) Update settings for Mattermost 4.7.2

## v1.6.3 (2018.01.29)

- [#18](https://github.com/ist-dsi/mattermost-cookbook/issues/18) Update settings for Mattermost 4.6.0

## v1.6.2 (2017.12.18)

- [#17](https://github.com/ist-dsi/mattermost-cookbook/issues/17) Update settings for Mattermost 4.5.0

## v1.6.1 (2017.12.15)

- [#16](https://github.com/ist-dsi/mattermost-cookbook/issues/16) Update settings for Mattermost 4.4.5

## v1.6.0 (2017.12.11)

- [#14](https://github.com/ist-dsi/mattermost-cookbook/issues/14) Update settings for Mattermost 4.4.3
- [#12](https://github.com/ist-dsi/mattermost-cookbook//issues/12) Refactor creation of `SystemD` service
- Set capability `CAP_NET_BIND_SERVICE` for `./platform` as stated in [installation guide](https://docs.mattermost.com/install/install-rhel-71.html#configuring-tls-on-mattermost-server).

## v1.5.5 (2017.11.23)

- [#13](https://github.com/ist-dsi/mattermost-cookbook/issues/13) Update settings for Mattermost 4.4.2

## v1.5.4 (2017.11.17)

- [#11](https://github.com/ist-dsi/mattermost-cookbook/issues/11) Update settings for Mattermost 4.4.1

## v1.5.3 (2017.11.13)

- [#10](https://github.com/ist-dsi/mattermost-cookbook/issues/10) Update settings for Mattermost 4.3.2

## v1.5.2 (2017.10.23)

- [#8](https://github.com/ist-dsi/mattermost-cookbook/issues/8) Update settings for Mattermost 4.3.1

## v1.5.0 (2017.10.16)

- [#6](https://github.com/ist-dsi/mattermost-cookbook/issues/6) Update settings for Mattermost 4.3.0

## v1.4.0 (2017.09.22)

- [#3](https://github.com/ist-dsi/mattermost-cookbook/issues/3) Update settings for Mattermost 4.2.0

## v1.3.0 (2017-08-29)

- Update settings for Mattermost 4.1.0

## v1.2.0 (2017-08-28)

- Update settings for Mattermost 4.0.4

## v1.1.0 (2017-08-28)

- Update settings for Mattermost 3.10.3

## v1.0.0 (2017-08-28)

- Update settings for Mattermost 3.9.2
- Add license file
- Refactor settings
- Fix linter errors

spec/spec_helper.rb
require 'chefspec'
require 'chefspec/berkshelf'

RSpec.configure do |config|
  config.platform = 'ubuntu'
  config.version = '14.04'
  config.log_level = :warn
end

at_exit { ChefSpec::Coverage.report! }


spec/default_spec.rb
require 'spec_helper'

describe 'mattermost-cookbook::default' do
  let(:chef_run) { ChefSpec::SoloRunner.new(platform: 'ubuntu', version: '16.04').converge(described_recipe) }

  it 'creates user mattermost' do
    expect(chef_run).to create_user('mattermost')
  end

  it 'downloads and extracts' do
    expect(chef_run).to extract_tar_extract('https://releases.mattermost.com/5.26.2/mattermost-5.26.2-linux-amd64.tar.gz').with(
      checksum: 'aa671915f684d7daca2fed321ede89dd05c3d377b0beaaca9561b1d7e3c36970',
      target_dir: '/opt',
      user: 'mattermost',
      group: 'mattermost'
    )
  end

  it 'creates data directory ' do
    expect(chef_run).to create_directory('/opt/mattermost/data').with(
      user: 'mattermost',
      mode: '755',
      recursive: true
    )
  end

  it 'creates plugins directory ' do
    expect(chef_run).to create_directory('/opt/mattermost/client/plugins').with(
      user: 'mattermost',
      mode: '755',
      recursive: true
    )
  end

  it 'creates mattermost config template' do
    expect(chef_run).to create_template('/opt/mattermost/config/config.json')
  end

  it 'enables and starts mattermost service' do
    expect(chef_run).to create_systemd_unit('mattermost.service')
    expect(chef_run).to enable_systemd_unit('mattermost.service')
    expect(chef_run).to start_systemd_unit('mattermost.service')
  end
end


spec/unit/recipes/default_spec.rb
require 'spec_helper'

describe 'mattermost-cookbook::default' do
  let(:chef_run) { ChefSpec::SoloRunner.new(platform: 'ubuntu', version: '16.04').converge(described_recipe) }

  it 'creates user mattermost' do
    expect(chef_run).to create_user('mattermost')
  end

  it 'downloads and extracts ark' do
    expect(chef_run).to put_ark('mattermost').with(
      url: 'https://releases.mattermost.com/3.5.0/mattermost-enterprise-3.5.0-linux-amd64.tar.gz',
      checksum: '0aa376254b74c32672118e304dd931d507968209f44cb5ca4099c8cc5b511699',
      path: '/opt',
      owner: 'mattermost'
    )
  end

  it 'creates data directory ' do
    expect(chef_run).to create_directory('/mattermost/data').with(
      user: 'mattermost',
      mode: 0755,
      recursive: true
    )
  end

  it 'creates mattermost config template' do
    expect(chef_run).to create_template('/opt/mattermost/config/config.json')
  end

  it 'creates mattermost init template' do
    expect(chef_run).to create_template('/etc/init/mattermost.conf')
  end

  it 'enables and starts mattermost service' do
    expect(chef_run).to enable_service('mattermost')
    expect(chef_run).to start_service('mattermost')
  end
end


recipes/default.rb
#
# Cookbook:: mattermost
# Recipe:: default
#
# Copyright:: (c) 2017 The Authors, All Rights Reserved.
install_directory = "#{node['mattermost']['config']['install_path']}/mattermost"

apt_package 'libcap2-bin' if platform_family?('debian')

user node['mattermost']['config']['user'] do
  action :create
end

directory install_directory do
  owner node['mattermost']['config']['user']
  group node['mattermost']['config']['user']
  mode '755'
  recursive true
  action :create
end

tar_extract node['mattermost']['package']['url'] do
  download_dir node['mattermost']['config']['install_path']
  target_dir node['mattermost']['config']['install_path']
  checksum node['mattermost']['package']['checksum']
  user node['mattermost']['config']['user']
  group node['mattermost']['config']['user']
  creates "#{install_directory}/config"
  action :extract
end

directory node['mattermost']['config']['data_dir'] do
  owner node['mattermost']['config']['user']
  group node['mattermost']['config']['user']
  mode '755'
  recursive true
  action :create
end

directory node['mattermost']['app']['file_settings']['directory'] do
  owner node['mattermost']['config']['user']
  group node['mattermost']['config']['user']
  mode '755'
  recursive true
  action :create
end

directory "#{install_directory}/#{node['mattermost']['app']['plugin_settings']['client_directory']}" do
  owner node['mattermost']['config']['user']
  group node['mattermost']['config']['user']
  mode '755'
  recursive true
  action :create
end

directory node['mattermost']['app']['log_settings']['file_location'] do
  owner node['mattermost']['config']['user']
  group node['mattermost']['config']['user']
  mode '755'
  recursive true
  action :create
end

directory node['mattermost']['app']['notification_log_settings'] do
  owner node['mattermost']['config']['user']
  group node['mattermost']['config']['user']
  mode '755'
  recursive true
  action :create
end

template "#{install_directory}/config/config.json" do
  source 'config.json.erb'
  owner node['mattermost']['config']['user']
  group node['mattermost']['config']['user']
  mode '0640'
  notifies :restart, 'systemd_unit[mattermost.service]'
end

execute "setcap cap_net_bind_service=+ep #{install_directory}/bin/platform" do
  user 'root'
end

systemd_unit 'mattermost.service' do
  content(
    Unit: {
      Description: 'Mattermost',
      After: node['mattermost']['systemd']['after'].join(' '),
    },
    Service: {
      ExecStart: "#{install_directory}/bin/mattermost",
      WorkingDirectory: install_directory,
      Restart: 'always',
      StandardOutput: 'syslog',
      StandardError: 'syslog',
      SyslogIdentifier: 'mattermost',
      User: node['mattermost']['config']['user'].to_s,
      Group: node['mattermost']['config']['group'].to_s,
    },
    Install: {
      WantedBy: 'multi-user.target',
    }
  )
  verify false
  action [:create, :enable, :start]
end


attributes/app.rb
default['mattermost']['app']['service_settings'] = {
  'site_url' => '',
  'websocket_url' => '',
  'license_file_location' => '',
  'listen_address' => ':8065',
  'connection_security' => '',
  'tls_cert_file' => '',
  'tls_key_file' => '',
  'tls_min_ver' => '1.2',
  'tls_strict_transport' => false,
  'tls_strict_transport_max_age' => 63072000,
  'tls_overwrite_ciphers' => [],
  'use_lets_encrypt' => false,
  'lets_encrypt_certificate_cache_file' => './config/letsencrypt.cache',
  'forward_80_to_443' => false,
  'read_timeout' => 300,
  'write_timeout' => 300,
  'maximum_login_attempts' => 10,
  'goroutine_health_threshold' => -1,
  'google_developer_key' => '',
  'enable_oauth_service_provider' => false,
  'enable_incoming_webhooks' => true,
  'enable_outgoing_webhooks' => true,
  'enable_commands' => true,
  'enable_only_admin_integrations' => true,
  'enable_post_username_override' => false,
  'enable_post_icon_override' => false,
  'enable_api_v3' => false,
  'enable_link_previews' => true,
  'enable_testing' => false,
  'enable_developer' => false,
  'enable_security_fix_alert' => true,
  'enable_insecure_outgoing_connections' => false,
  'enable_untrusted_internal_connections' => '',
  'enable_multifactor_authentication' => false,
  'enforce_multifactor_authentication' => false,
  'enable_user_access_tokens' => false,
  'allow_cors_from' => '',
  'cors_exposed_headers' => '',
  'cors_allow_credentials' => false,
  'cors_debug' => false,
  'allow_cookies_for_subdomains' => false,
  'session_length_web_in_days' => 30,
  'session_length_mobile_in_days' => 30,
  'session_length_sso_in_days' => 30,
  'session_cache_in_minutes' => 10,
  'session_idle_timeout_in_minutes' => 0,
  'websocket_secure_port' => 443,
  'websocket_port' => 80,
  'webserver_mode' => 'gzip',
  'enable_custom_emoji' => false,
  'enable_custom_picker' => true,
  'enable_gif_picker' => false,
  'gfycat_api_key' => '2_KtH_W5',
  'gfycat_api_secret' => '3wLVZPiswc3DnaiaFoLkDvB4X0IV6CpMkj4tf2inJRsBY6-FnkT08zGmppWFgeof',
  'restrict_custom_emoji_creation' => 'all',
  'restrict_post_delete' => 'all',
  'allow_edit_post' => 'always',
  'post_edit_time_limit' => -1,
  'time_between_user_typing_updates_milliseconds' => 5000,
  'enable_post_search' => true,
  'enable_user_typing_messages' => true,
  'enable_channel_viewed_messages' => true,
  'enable_user_statuses' => true,
  'experimental_enable_authentication_transfer' => true,
  'cluster_log_timeout_milliseconds' => 2000,
  'close_unused_direct_messages' => false,
  'enable_preview_features' => true,
  'enable_tutorial' => true,
  'experimental_enable_default_channel_leave_join_messages' => true,
  'experimental_group_unread_channels' => 'disabled',
  'group_filter' => '',
  'group_display_name_attribute' => '',
  'group_id_attribute' => '',
  'enable_api_team_deletion' => false,
  'experimental_enable_hardened_mode' => false,
  'experimental_limit_client_config' => false,
  'enable_email_invitations' => false,
  'disable_legacy_mfa' => false,
  'minimum_hashtag_length' => 3,
  'enable_bot_account_creation' => false,
  'disable_bots_when_owner_is_deactivated' => true,
  'trusted_proxy_ip_header' => [],
  'enable_latex' => true,
  'enable_open_tracing' => false,
  'idle_timeout' => 60,
  'experimental_channel_sidebar_organization' => 'off',
  'extend_session_length_with_activity' => true,
  'enable_local_mode' => false,
  'local_mode_socket_location' => '/var/tmp/mattermost_local.socket',
  'experimental_data_prefetch' => true,
}
default['mattermost']['app']['image_proxy_settings'] = {
  'enable_image_proxy' => true,
  'image_proxy_type' => 'local',
  'remote_image_proxy_url' => '',
  'remote_image_proxy_options' => '',
}

default['mattermost']['app']['team_settings'] = {
  'site_name' => 'Mattermost',
  'max_users_per_team' => 50,
  'enable_team_creation' => true,
  'enable_user_creation' => true,
  'enable_open_server' => false,
  'enable_user_deactivation' => false,
  'restrict_creation_to_domains' => '',
  'enable_custom_brand' => false,
  'custom_brand_text' => '',
  'custom_description_text' => '',
  'restrict_direct_message' => 'any',
  'restrict_team_invite' => 'all',
  'restrict_public_channel_management' => 'all',
  'restrict_private_channel_management' => 'all',
  'restrict_public_channel_creation' => 'all',
  'restrict_private_channel_creation' => 'all',
  'restrict_public_channel_deletion' => 'all',
  'restrict_private_channel_deletion' => 'all',
  'restrict_private_channel_manage_members' => 'all',
  'enable_x_to_leave_channels_from_lhs' => false,
  'user_status_away_timeout' => 300,
  'max_channels_per_team' => 2000,
  'max_notifications_per_channel' => 1000,
  'enable_confirm_notifications_to_channel' => true,
  'teammate_name_display' => 'username',
  'experimental_view_archived_channels' => false,
  'experimental_default_channels' => '',
  'experimental_enable_automatic_replies' => false,
  'experimental_hide_town_square_in_lhs' => false,
  'experimental_town_square_is_read_only' => false,
  'experimental_primary_team' => '',
  'lock_teammate_name_display' => [],
}

default['mattermost']['app']['client_requirements'] = {
  'android_latest_version' => '',
  'android_min_version' => '',
  'desktop_latest_version' => '',
  'desktop_min_version' => '',
  'ios_latest_version' => '',
  'ios_min_version' => '',
}

default['mattermost']['app']['sql_settings'] = {
  'driver_name' => 'mysql',
  'address' => 'localhost',
  'port' => '3306',
  'username' => 'mmuser',
  'password' => 'mostest',
  'name' => 'mattermost_test',
  'data_source_replicas' => [],
  'data_source_search_replicas' => [],
  'conn_max_lifetime_milliseconds' => 3600000,
  'max_idle_conns' => 20,
  'max_open_conns' => 300,
  'trace' => false,
  'at_rest_encrypt_key' => '',
  'query_timeout' => 30,
  'enable_public_channels_materialization' => true,
  'disable_database_search' => false,
}

default['mattermost']['app']['notification_log_settings'] = {
  'enable_console' => true,
  'console_level' => 'DEBUG',
  'console_json' => true,
  'enable_file' => true,
  'file_level' => 'INFO',
  'file_json' => true,
  'file_location' => '/var/log/mattermost',
}

default['mattermost']['app']['log_settings'] = {
  'enable_console' => true,
  'console_level' => 'DEBUG',
  'console_json' => true,
  'enable_file' => true,
  'file_level' => 'INFO',
  'file_json' => true,
  'file_location' => '/var/log/mattermost',
  'enable_webhook_debugging' => true,
  'enable_diagnostics' => true,
}

default['mattermost']['app']['password_settings'] = {
  'minimum_length' => 5,
  'lowercase' => false,
  'number' => false,
  'uppercase' => false,
  'symbol' => false,
}

default['mattermost']['app']['file_settings'] = {
  'enable_file_attachments' => true,
  'enable_mobile_upload' => true,
  'enable_mobile_download' => true,
  'max_file_size' => 52428800,
  'driver_name' => 'local',
  'directory' => './data/',
  'enable_public_link' => false,
  'public_link_salt' => '', # SET THIS!
  'initial_font' => 'luximbi.ttf',
  'amazon_s3_access_key_id' => '',
  'amazon_s3_secret_access_key' => '',
  'amazon_s3_bucket' => '',
  'amazon_s3_region' => '',
  'amazon_s3_endpoint' => 's3.amazonaws.com',
  'amazon_s3_ssl' => true,
  'amazon_s3_sign_v2' => false,
  'amazon_s3_sse' => false,
  'amazon_s3_trace' => false,
  'amazon_S3_path_prefix' => '',
}

default['mattermost']['app']['email_settings'] = {
  'enable_sign_up_with_email' => true,
  'enable_sign_in_with_email' => true,
  'enable_sign_in_with_username' => true,
  'send_email_notifications' => true,
  'use_channel_in_email_notifications' => false,
  'require_email_verification' => false,
  'feedback_name' => '',
  'feedback_email' => 'test@example.com',
  'feedback_organization' => '',
  'enable_smtp_auth' => false,
  'smtp_username' => '',
  'smtp_password' => '',
  'smtp_server' => 'localhost',
  'smtp_server_timeout' => 10,
  'smtp_port' => '2500',
  'connection_security' => '',
  'invite_salt' => '', # SET THIS!
  'send_push_notifications' => false,
  'push_notification_server' => '',
  'push_notification_contents' => 'generic',
  'push_notification_buffer' => 1000,
  'enable_email_batching' => false,
  'email_batching_buffer_size' => 256,
  'email_batching_interval' => 30,
  'enable_preview_mode_banner' => true,
  'skip_server_certificate_verification' => false,
  'email_notification_contents_type' => 'full',
  'login_button_color' => '',
  'login_button_border_color' => '',
  'login_button_text_color' => '',
}

default['mattermost']['app']['rate_limit_settings'] = {
  'enable' => false,
  'per_sec' => 10,
  'max_burst' => 100,
  'memory_store_size' => 10000,
  'vary_by_remote_addr' => true,
  'vary_by_user' => false,
  'vary_by_header' => '',
}

default['mattermost']['app']['privacy_settings'] = {
  'show_email_address' => true,
  'show_full_name' => true,
}

default['mattermost']['app']['support_settings'] = {
  'terms_of_service_link' => 'https://about.mattermost.com/default-terms/',
  'privacy_policy_link' => 'https://about.mattermost.com/default-privacy-policy/',
  'about_link' => 'https://about.mattermost.com/default-about/',
  'help_link' => 'https://about.mattermost.com/default-help/',
  'report_a_problem_link' => 'https://about.mattermost.com/default-report-a-problem/',
  'support_email' => 'feedback@mattermost.com',
}

default['mattermost']['app']['announcement_settings'] = {
  'enable_banner' => false,
  'banner_text' => '',
  'banner_color' => '#f2a93b',
  'banner_text_color' => '#333333',
  'allow_banner_dismissal' => true,
}

default['mattermost']['app']['theme_settings'] = {
  'enable_theme_selection' => true,
  'default_theme' => 'default',
  'allow_custom_themes' => true,
  'allowed_themes' => [],
}

default['mattermost']['app']['gitlab_settings'] = {
  'enable' => false,
  'secret' => '',
  'id' => '',
  'scope' => '',
  'auth_endpoint' => '',
  'token_endpoint' => '',
  'user_api_endpoint' => '',
}

default['mattermost']['app']['google_settings'] = {
  'enable' => false,
  'secret' => '',
  'id' => '',
  'scope' => 'profile email',
  'auth_endpoint' => 'https://accounts.google.com/o/oauth2/v2/auth',
  'token_endpoint' => 'https://www.googleapis.com/oauth2/v4/token',
  'user_api_endpoint' => 'https://www.googleapis.com/plus/v1/people/me',
}

default['mattermost']['app']['office_365_settings'] = {
  'enable' => false,
  'secret' => '',
  'id' => '',
  'scope' => 'User.Read',
  'auth_endpoint' => 'https://login.microsoftonline.com/common/oauth2/v2.0/authorize',
  'token_endpoint' => 'https://login.microsoftonline.com/common/oauth2/v2.0/token',
  'user_api_endpoint' => 'https://graph.microsoft.com/v1.0/me',
}

default['mattermost']['app']['ldap_settings'] = {
  'enable' => false,
  'enable_sync' => false,
  'ldap_server' => '',
  'ldap_port' => 389,
  'connection_security' => '',
  'base_dn' => '',
  'bind_username' => '',
  'bind_password' => '',
  'user_filter' => '',
  'first_name_attribute' => '',
  'last_name_attribute' => '',
  'email_attribute' => '',
  'username_attribute' => '',
  'nickname_attribute' => '',
  'id_attribute' => '',
  'position_attribute' => '',
  'login_id_attribute' => '',
  'sync_interval_minutes' => 60,
  'skip_certificate_verification' => false,
  'query_timeout' => 60,
  'max_page_size' => 0,
  'login_field_name' => '',
  'login_button_color' => '',
  'login_button_border_color' => '',
  'login_button_text_color' => '',
  'guest_filter' => '',
  'enable_admin_filter' => false,
  'admin_filter' => '',
  'picture_attribute' => '',
}

default['mattermost']['app']['compliance_settings'] = {
  'enable' => false,
  'directory' => './data/',
  'enable_daily' => false,
}

default['mattermost']['app']['localization_settings'] = {
  'default_server_locale' => 'en',
  'default_client_locale' => 'en',
  'available_locales' => '',
}

default['mattermost']['app']['saml_settings'] = {
  'enable' => false,
  'enable_sync_with_ldap' => false,
  'enable_sync_with_ldap_include_auth' => false,
  'verify' => true,
  'encrypt' => true,
  'idp_url' => '',
  'idp_descriptor_url' => '',
  'assertion_consumer_service_url' => '',
  'scoping_idp_provider_id' => '',
  'scoping_idp_name' => '',
  'idp_certificate_file' => '',
  'public_certificate_file' => '',
  'private_key_file' => '',
  'id_attribute' => '',
  'first_name_attribute' => '',
  'last_name_attribute' => '',
  'email_attribute' => '',
  'username_attribute' => '',
  'nickname_attribute' => '',
  'locale_attribute' => '',
  'position_attribute' => '',
  'login_button_text' => 'With SAML',
  'login_button_color' => '',
  'login_button_border_color' => '',
  'login_button_text_color' => '',
  'signature_algorithm' => '',
  'canonical_algorithm' => '',
  'guest_attribute' => '',
  'idp_metadata_url' => '',
  'enable_admin_attribute' => true,
  'admin_attribute' => '',
  'service_provider_identifier' => '',
}

default['mattermost']['app']['native_app_settings'] = {
  'app_download_link' => 'https://mattermost.com/download/#mattermostApps',
  'android_app_download_link' => 'https://about.mattermost.com/mattermost-android-app/',
  'ios_app_download_link' => 'https://about.mattermost.com/mattermost-ios-app/',
}

default['mattermost']['app']['cluster_settings'] = {
  'enable' => false,
  'cluster_name' => '',
  'override_hostname' => '',
  'use_ip_address' => true,
  'use_experimental_gossip' => false,
  'read_only_config' => true,
  'gossip_port' => 8074,
  'streaming_port' => 8075,
  'max_idle_conns' => 100,
  'max_idle_conns_per_host' => 128,
  'idle_conn_timeout_milliseconds' => 90000,
  'network_interface' => '',
  'bind_address' => '',
  'advertise_address' => '',
  'enable_experimental_gossip_encryption' => false,
}

default['mattermost']['app']['metrics_settings'] = {
  'enable' => false,
  'block_profile_rate' => 0,
  'listen_address' => ':8067',
}

default['mattermost']['app']['experimental_settings'] = {
  'restrict_system_admin' => false,
  'client_side_cert_enable' => false,
  'client_side_cert_check' => 'secondary',
  'disable_post_metadata' => false,
  'link_metadata_timeout_milliseconds' => 5000,
}

default['mattermost']['app']['analytics_settings']['max_users_for_statistics'] = 2500

default['mattermost']['app']['webrtc_settings'] = {
  'enable' => false,
  'gateway_websocket_url' => '',
  'gateway_admin_url' => '',
  'gateway_admin_secret' => '',
  'stun_uri' => '',
  'turn_uri' => '',
  'turn_username' => '',
  'turn_shared_key' => '',
}

default['mattermost']['app']['elastic_search_settings'] = {
  'connection_url' => 'http://localhost:9200',
  'username' => 'elastic',
  'password' => 'changeme',
  'enable_indexing' => false,
  'enable_searching' => false,
  'skip_tls_verification' => false,
  'sniff' => true,
  'post_index_replicas' => 1,
  'post_index_shards' => 1,
  'aggregate_posts_after_days' => 365,
  'post_aggregator_job_start_time' => '03:00',
  'index_prefix' => '',
  'live_indexing_batch_size' => 1,
  'request_timeout_seconds' => 30,
  'bulk_indexing_time_window_seconds' => 3600,
}

default['mattermost']['app']['data_retention_settings'] = {
  'enable_message_deletion' => false,
  'enable_file_deletion' => false,
  'message_retention_days' => 365,
  'file_retention_days' => 365,
  'deletion_job_start_time' => '02:00',
}

default['mattermost']['app']['message_export_settings'] = {
  'enable_export' => false,
  'daily_run_time' => '01:00',
  'export_from_timestamp' => 0,
  'file_location' => 'export',
  'batch_size' => 10000,
  'global_relay_settings' => {
    'customer_type' => 'A9',
    'smtp_username' => '',
    'smtp_password' => '',
    'email_address' => '',
  },
}

default['mattermost']['app']['job_settings'] = {
  'run_jobs' => true,
  'run_scheduler' => true,
}

default['mattermost']['app']['plugin_settings'] = {
  'enable' => true,
  'enable_uploads' => false,
  'directory' => 'plugins',
  'client_directory' => 'client/plugins',
  'enable_health_check' => true,
  'plugins' => {},
  'plugin_states' => {},
  'enable_market_place' => true,
  'market_place_url' => 'https://marketplace.integrations.mattermost.com',
  'require_plugin_signature' => true,
  'signature_public_keyfiles' => {},
  'enable_remote_marketplace' => true,
  'automatic_prepackaged_plugins' => true,
}

default['mattermost']['app']['compliance_Settings'] = {
  'sign_request' => '',
}

default['mattermost']['app']['guest_accounts_settings'] = {
  'enable' => false,
  'allow_email_accounts' => true,
  'enforce_multifactor_authentication' => false,
  'restrict_creation_to_domains' => true,
}

default['mattermost']['app']['bleve_settings'] = {
  'index_dir' => '',
  'enable_indexing' => false,
  'enable_searching' => false,
  'enable_autocomplete' => false,
  'bulk_indexing_time_window_seconds' => 3600,
}
